src/hcl/parser.py

```diff
@@ -72,9 +72,14 @@ def objectlist_flat(self, lt, replace):
                 if isinstance(v, dict):
                     dd = d.setdefault(k, {})
                     for kk, vv in iteritems(v):
-                        if kk in dd.keys():
-                            for k2, v2 in iteritems(vv):
-                                dd[kk][k2] = v2
+                        if type(dd) == list:
+                            dd.append(vv)
+                        elif kk in dd.keys():
+                            if hasattr(vv, 'iteritems') or hasattr(vv, 'items'):
```

Comment:
You could check just for 'items', it would be slightly more efficient (as items exists on py2 and py3).